common/auth/tls.go

```diff
@@ -0,0 +1,42 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package auth
+
+type (
+	// TLS describe TLS configuration (for Kafka, Cassandra)
+	TLS struct {
+		Enabled bool `yaml:"enabled"`
+
+		// CertPath and KeyPath are optional depending on server
+		// config, but both fields must be omitted to avoid using a
+		// client certificate
+		CertFile string `yaml:"certFile"`
+		KeyFile  string `yaml:"keyFile"`
+
+		BundleFile string `yaml:"bundleFile"`
+
+		CaFile string `yaml:caFile` //optional depending on server config
```

Comment:
these 2 variables are the same. let us keep "CaFile string `yaml:caFile`" for readability.
but before landing this, let me check with one customer who is already using TLS with kafka

Reply:
plz also do a `make fmt` locally

Reply:
I ran `make fmt` earlier, and it made some formatting changes (included in the last commit).
I just ran it again with no change. Is there something you expect to be reformatted?

Reply:
one more thing, the `Contributor License Agreement` does not really recognize your name. one thing you could try is to squash all commits and force push

Reply:
> I ran `make fmt` earlier, and it made some formatting changes (included in the last commit).
> I just ran it again with no change. Is there something you expect to be reformatted?

if that is the case, then plz just ignore

Reply:
> one more thing, the `Contributor License Agreement` does not really recognize your name. one thing you could try is to squash all commits and force push

I think because I made the first commit from my work account, which is private. I tried to re-author it but that apparently didn't work. I'll try the squash & force push...